### DISCUSSION TITLE
Switching Virtual Desktops Causes Taskbar Labels to Collapse

### DISCUSSION THREAD

**Discussion by PreXident** [ramensoftware/windhawk-mods#397] [Q&A] [answered]
Hello, I recently got a strange inconsistency when switching virtual desktops with Taskbar Labels for Windows 11, the taskbar labels collapse and stay that way until a new window is opened.

Before switching:
![pic1](https://github.com/ramensoftware/windhawk-mods/assets/5681128/bf4096c4-150a-4738-bf53-210441989df4)
After switching to secondary desktop and back:
![pic2](https://github.com/ramensoftware/windhawk-mods/assets/5681128/1467439b-eab6-424b-aa3c-869747c23faf)

I also use a few other mods for taskbar:

- Disable grouping on the taskbar
- Middle click to close on the taskbar
- Taskbar Clock Customization
- Taskbar height and icon size
- Virtual Desktop Preserve Taskbar Order

I did a few experiments and it seems there is a new (or at least unknown to me) option in Taskbar Panel to never combine items in the taskbar, but it does not seem to be working for me and causes strange icon flickering, so I need to set it to always anyway.

Maybe I just use a strange combination of mods and workflow? Thank you for any tips!

**Comment by m417z** [comment]
Should be fixed in version 1.2 of the mod. Please try it out and let me know whether it works.

**Comment by PreXident** [comment]
Hello, it works very nice with one exception: the pinned items on the taskbar also have labels now.

No program running:
![taskbar1](https://github.com/ramensoftware/windhawk-mods/assets/5681128/ef311545-b961-419b-9f7d-06d8adac28d3)

Firefox and Windhawk running:
![taskbar2](https://github.com/ramensoftware/windhawk-mods/assets/5681128/44881075-0f20-453d-bf0f-28b3fbb9c463)

Original behaviour:
![pic1](https://github.com/ramensoftware/windhawk-mods/assets/5681128/c4f3a2ee-071e-42a5-b827-efa5ff612dfa)

Would it be possible to add an option to settings to have pinned items without labels, please? I use them as a replacement for Taskbar Toolbar feature that is unfortunately missing in Windows 11. Thank you very much!

**Reply by m417z** [reply, marked as answer]
Should be fixed in 1.2.1 which was released shortly after 1.2.
